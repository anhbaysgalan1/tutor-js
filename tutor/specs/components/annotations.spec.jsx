import AnnotationWidget from '../../src/components/annotations/annotation';
import Renderer from 'react-test-renderer';
import { bootstrapCoursesList } from '../courses-test-data';
import AnnotationsMap from '../../src/models/annotations';
import User from '../../src/models/user';
import Page from '../../api/pages/be8818d0-2dba-4bf3-859a-737c25fb2c99@20.json';
import ANNOTATIONS from '../../api/annotations.json';

jest.mock('../../src/models/feature_flags', () => ({ is_highlighting_allowed: true }));
jest.mock('../../src/models/user');

describe('Annotations', () => {

  let props, Courses, body, annotations;

  beforeEach(function() {
    Courses = bootstrapCoursesList();
    Courses.get(1).appearance_code = 'college_biology';
    body = window.document.body;
    annotations = new AnnotationsMap();
    User.annotations = annotations;
    annotations.updateAnnotations(
      ANNOTATIONS.rows
    );
    //mockAnnotationsMap.updateAnnotations(
    // from reference_book/page component
    body.innerHTML = '<div id="mount"><div class="book-content">' +
      Page.content_html
      .replace(/^[\s\S]*<body[\s\S]*?>/, '')
      .replace(/<\/body>[\s\S]*$/, '') +
      '</div></div>';
    window.getSelection = jest.fn(() => ({
      removeAllRanges: jest.fn(),
      addRange: jest.fn(),
      getRangeAt: jest.fn(() => ({
        cloneRange: jest.fn(),
        getBoundingClientRect: jest.fn(() => ({
          bottom: 150,
          top: 100,
          left: 100,
          right: 150,
        })),
      })),
    }));
    window.document.createRange = jest.fn(() => ({
      setStart: jest.fn(),
      setEnd: jest.fn(),
      collapse: jest.fn(),
    }));
    props = {
      courseId: '1',
      documentId: 'be8818d0-2dba-4bf3-859a-737c25fb2c99',
      title: Page.title,
      windowImpl: window,
      chapter: Page.chapter_section[0],
      section: Page.chapter_section[1],
    };
  });

  it('sorts in model', () => {
    expect(Object.keys(annotations.byCourseAndPage)).toEqual(['1']);
    expect(Object.keys(annotations.byCourseAndPage[1])).toEqual(['2.1']);
    expect(Object.keys(annotations.byCourseAndPage[1]['2.1'])).toHaveLength(2);
  });

  it('renders and matches snapshot', () => {
    const comp = Renderer.create(<AnnotationWidget {...props} />);
    expect(comp.toJSON()).toMatchSnapshot();
    comp.unmount();
  });
});
