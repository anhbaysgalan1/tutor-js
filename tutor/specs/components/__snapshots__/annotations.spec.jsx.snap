// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Annotations renders and matches snapshot 1`] = `
<div
  className="annotater"
>
  <div
    className="slide-out-edit-box closed"
  />
  <div
    className="annotation-edit-buttons"
  >
    <i
      alt="View annotation"
      className="tutor-icon fa fa-comment sidebar-button clickable"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "top": NaN,
        }
      }
      type="comment"
    />
  </div>
  <div
    className="status-message-toast info"
  >
    <i
      className="tutor-icon fa fa-info-circle"
      role="presentation"
      type="info-circle"
    />
     
    Waiting for page to finish loadingâ€¦
  </div>
  <div
    className="highlights-windowshade up"
  >
    <div
      className="centered-content"
    >
      <div
        className="summary-page"
      >
        <h1>
          Highlights and annotations
        </h1>
        <div
          className="filter-area"
        >
          <div
            className="filter-widget"
          >
            <div
              className="multi-select"
            >
              <div
                className="dropdown btn-group"
              >
                <button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="dropdown-toggle btn btn-default"
                  disabled={false}
                  id="multi-select"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                  type="button"
                >
                  Display sections
                   
                  <span
                    className="caret"
                  />
                </button>
                <ul
                  aria-labelledby="multi-select"
                  className="dropdown-menu"
                  role="menu"
                >
                  <li
                    className="multi-selection-option"
                    role="presentation"
                    style={undefined}
                  >
                    <a
                      href="#"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      role="menuitem"
                      tabIndex="-1"
                    >
                      <i
                        className="tutor-icon fa fa-check-square-o"
                        role="presentation"
                        type="check-square-o"
                      />
                      <span
                        className="title"
                      >
                        <span>
                          <span
                            className="section"
                          >
                            2.1
                          </span>
                          <span>
                            Atoms, Isotopes, Ions, and Molecules: The Building Blocks
                          </span>
                        </span>
                      </span>
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div>
            <button
              className="print-btn btn btn-default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <i
                className="tutor-icon fa fa-print"
                role="presentation"
                type="print"
              />
               Print this page
            </button>
          </div>
        </div>
        <div
          className="annotations"
        >
          <div>
            <h2>
              2.1
               
              Atoms, Isotopes, Ions, and Molecules: The Building Blocks
            </h2>
            <div
              className="annotation-card"
            >
              <div
                className="annotation-content"
              >
                <blockquote
                  className="selected-text"
                >
                  ng world, elements are found in different proportions, and some elements common to living organisms are relatively rare on the earth as a whole, as shown in See Table. For examp
                </blockquote>
                <div
                  className="plain-text"
                >
                  ee
                </div>
              </div>
              <div
                className="controls"
              >
                <button
                  onClick={[Function]}
                  title="Edit"
                >
                  <i
                    className="tutor-icon fa fa-edit"
                    role="presentation"
                    type="edit"
                  />
                </button>
                <button
                  onClick={[Function]}
                  title="View in book"
                >
                  <i
                    className="tutor-icon fa fa-external-link"
                    role="presentation"
                    type="external-link"
                  />
                </button>
                <button
                  onClick={[Function]}
                  title="Delete"
                >
                  <i
                    className="tutor-icon fa fa-trash"
                    role="presentation"
                    type="trash"
                  />
                </button>
              </div>
            </div>
            <div
              className="annotation-card"
            >
              <div
                className="annotation-content"
              >
                <blockquote
                  className="selected-text"
                >
                  Hydrogen (H)
                10%
                trace
                0.1%
                </blockquote>
                <div
                  className="plain-text"
                >
                  
                </div>
              </div>
              <div
                className="controls"
              >
                <button
                  onClick={[Function]}
                  title="Edit"
                >
                  <i
                    className="tutor-icon fa fa-edit"
                    role="presentation"
                    type="edit"
                  />
                </button>
                <button
                  onClick={[Function]}
                  title="View in book"
                >
                  <i
                    className="tutor-icon fa fa-external-link"
                    role="presentation"
                    type="external-link"
                  />
                </button>
                <button
                  onClick={[Function]}
                  title="Delete"
                >
                  <i
                    className="tutor-icon fa fa-trash"
                    role="presentation"
                    type="trash"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
